lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -67,3 +75,20 @@ data ErrPostTx
     deriving (Generic, Show, Eq)
 
 instance Exception ErrPostTx
+
+-- | Tries to waits 20 s, until 'networkTip networkLayer' succeeds.
+waitForConnection
+    :: NetworkLayer IO
+    -> Int -- ^ Number of retries / timeout in seconds
```

Comment:
This is exactly what `Quantity` is for! To convey this kind of information :wink: Every time you see yourself putting a comment to explain a `What` (and not a `Why`), ask yourself if you couldn't instead craft a better type that answers it.


Also, the comment seems to be a lie.

Reply:
> Also, the comment seems to be a lie.

Fair point.

> This is exactly what Quantity is for!

If we do `retry` it won't matter, but:

> This is exactly what Quantity is for!

Hmmmmmmm.… 🤔 

As long as we don't just
```
f (Quantity 1000000) 
```
sure, maybe.

Replacing the stock `threadDelay` with a typed version — everywhere — might be an idea too.